sql/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -143,13 +148,25 @@ public LogicalPlan planSubSelect(SelectSymbol selectSymbol, PlannerContext plann
         return planBuilder;
     }
 
+
+    public LogicalPlan normalizeAndPlan(QueriedRelation relation,
+                                        PlannerContext plannerContext,
+                                        SubqueryPlanner subqueryPlanner,
+                                        FetchMode fetchMode) {
+        return plan(
+            (QueriedRelation) relationNormalizer.normalize(relation, plannerContext.transactionContext()),
+            plannerContext,
+            subqueryPlanner,
+            fetchMode
+        );
+    }
+
     public LogicalPlan plan(QueriedRelation queriedRelation,
```

Comment:
is it still valid for this to be public?

Reply:
@andreidan In the insert-from-subquery case we normalize already up-front to be able to rewrite references to do source lookups. But I pushed a fixup so that we've a single public plan method that does the normalization. If the structure is already normalized it won't do much - if anything - anyways.

Reply:
sounds good